apps/src/templates/instructions/InstructionsWithWorkspace.jsx

```diff
@@ -110,14 +153,30 @@ export class UnwrappedInstructionsWithWorkspace extends React.Component {
 
 export default connect(
   function propsFromStore(state) {
+    const {hasContainedLevels, isEmbedView, isShareView} = state.pageConstants;
+    const {shortInstructions, longInstructions} = state.instructions;
+    const showInstructions = !!(
+      !isShareView &&
+      (shortInstructions || longInstructions || hasContainedLevels)
+    );
+    const showResizer = showInstructions && !isEmbedView;
```

Comment:
The fix for the bug in the last attempt is right here.  I've cleaned up the props for `InstructionsWithWorkspace` to be the more obvious `showInstructions` and `showResizer`, and embedded the related logic in the connect wrapper.  This logic is covered in integration tests, below.